### DISCUSSION TITLE
!!!How to decrypt encrypted data? I wrongly delete all my data in a vault!

### DISCUSSION THREAD

**Discussion by GaryRandom** [remotely-save/remotely-save#303] [General]
Hi, everyone. I have done a stupid wrong operation that delete all my data in my Mac.
Fortunately, I have a backup of remote-save encrypted data.
Now, I want to restore all my data from this backup.
How can I do this?
PLEASE HELP.

**Comment by fyears** [comment]
hi sorry for the late reply but i will leave a solution here if you or anyone needs it. actually i have considered this situation from the very beginning and created a doc.

1. make a backup of encrypted files again!
2. you can use openssl command line: <https://github.com/remotely-save/remotely-save/blob/master/docs/encryption.md>

```

# file content encryption (ignoring file path encryption)
openssl enc -p -aes-256-cbc -pbkdf2 -iter 20000 -pass pass:somepassword -in ./sometext.txt -out ./sometext.txt.enc

# file content decryption (ignoring file path decryption)
openssl enc -d -p -aes-256-cbc -pbkdf2 -iter 20000 -pass pass:somepassword -in ./sometext.txt.enc -out ./sometext.txt

```
